src/components/PostDetails.tsx

```diff
@@ -157,9 +88,7 @@ export const PostDetails: React.FC<Props> = ({ post }) => {
           </button>
         )}
 
-        {loaded && !hasError && visible && (
-          <NewCommentForm onSubmit={addComment} />
-        )}
+        {loaded && !hasError && visible && <NewCommentForm />}
```

Comment:
Consider creating a separate component for the NewCommentForm visibility logic. It will improve the maintainability of the code and reduce the complexity of this component.